src/securechip/securechip.c

```diff
@@ -764,3 +764,13 @@ bool securechip_u2f_counter_inc(uint32_t* counter)
     return _write_data_slot_block(&data.bytes[0], SECURECHIP_SLOT_DATA0, 0);
 }
 #endif
+
+bool securechip_model(securechip_model_t* model_out)
+{
+    uint8_t revision[4] = {0};
+    if (atcab_info(revision) != ATCA_SUCCESS) {
+        return false;
+    }
+    *model_out = revision[3] >= 0x03 ? SECURECHIP_ATECC608B : SECURECHIP_ATECC608A;
```

Comment:
I remember reading about somewhere but can't find where it says that any revision greater than 0x03 is 608B. @benma do you remember? Places I looked:

- http://ww1.microchip.com/downloads/en/Appnotes/Migrating-from-the-ATECC608A-to-the-ATECC608B-DS40002237A.pdf
- https://ww1.microchip.com/downloads/en/DeviceDoc/ATECC608B-CryptoAuthentication-Device-Summary-Data-Sheet-DS40002239A.pdf
- http://ww1.microchip.com/downloads/en/Appnotes/Atmel-8984-CryptoAuth-CryptoAuthLib-ApplicationNote.pdf
- https://microchiptech.github.io/cryptoauthlib/a01516.html#ga6e19ecd60be4c74665d9ec142e460771

Maybe I missed it somewhere in https://www.microchip.com/wwwproducts/en/ATECC608B.

Reply:
Ok, checked elsewhere offline. This is indeed the case: values starting with 0x03 and on indicate 608B. :+1: